cloudinit/config/cc_power_state_change.py

```diff
@@ -167,44 +155,16 @@ def load_power_state(cfg, distro_name):
     if not isinstance(pstate, dict):
         raise TypeError("power_state is not a dict.")
 
-    opt_map = {'halt': '-H', 'poweroff': '-P', 'reboot': '-r'}
-
+    modes_ok = ['halt', 'poweroff', 'reboot']
     mode = pstate.get("mode")
-    if mode not in opt_map:
+    if mode not in modes_ok:
```

Comment:
In case we ever end up with a distro that has a different set of modes, we may want to make `shutdown_options_map` the canonical source of this:

```suggestion
    if mode not in distro.shutdown_options_map:
```

What do you think?

Reply:
Yes, of course. I had overlooked that in my changes yesterday. Thanks for pointing that out.